### ISSUE TITLE
no-shadowed-variable false positive in function typedefs/ functions defs in interfaces

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#500]
the following code incorrectly reports a shadowed variable failure for `a` within the typedef of `done`:

```ts
function foo(a: any, done: (a: any) => void) {
    //
}
```

**Comment by JKillian** [comment]
This also happens in interface declarations, such as:

```typescript
interface Surface {
    diffuse: (pos: Vector) => Color;
    specular: (pos: Vector) => Color;
    reflect: (pos: Vector) => number;
    roughness: number;
}
```
I'm going to expand the scope of this issue to include this as well

**Comment by adidahiya** [comment]
:+1:

**Comment by adidahiya** [comment]
another false positive example:

```ts
function foo(bar: (a: any) => void) {
    let a = 1;
}
```

`a` is incorrectly reported as a shadowed variable
